runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -597,10 +607,58 @@ Table<K, V, M> openTable(@Nonnull final String namespace,
                 mapSupplier);
         tableMap.put(fullyQualifiedTableName, (Table<Message, Message, Message>) table);
 
+        resetIfTableTrimmed(namespace, tableName);
         registerTable(namespace, tableName, kClass, vClass, mClass, tableOptions);
         return table;
     }
 
+    /**
+     * If the table that is going to be registered, is not there in the registry table,
+     * but the address space seems to contain the stream map with no content, it means
+     * the table was trimmed without being checkpointing. If a trimmed table of this sort
+     * is found, append a HOLE to the stream to reset it.
+     *
+     * @param namespace Namespace of the table
+     * @param tableName Name of the table
+     */
+    private void resetIfTableTrimmed(String namespace, String tableName) {
+        TableName tableNameKey = TableName.newBuilder()
+                .setNamespace(namespace)
+                .setTableName(tableName)
+                .build();
+        CorfuRecord<TableDescriptors, TableMetadata> corfuRecord = this.registryTable.get(tableNameKey);
```

Comment:
You can check if corfuRecord is null right away to avoid sequencer query if it's indeed null.